modules/ipcscript.js

```diff
@@ -48,6 +50,21 @@ function IPCScript_getMetaStr() {
   return this._metaStr;
 });
 
+IPCScript.prototype.dedup = function() {
+  // scripts sent via e10s message manager contain new string instances every time a new one is created -> dedup them to reduce memory footprint
+
+  // alternative approaches:
+  // a) beam down IPC script instances only once via the parent process manager every time they change.
+  // b) creating resource URIs and loading user scripts straight from URIs might also be more elegant
+  //    see: https://developer.mozilla.org/en-US/docs/Mozilla/JavaScript_code_modules/Using#Programmatically_adding_aliases
+  var old = instances.get(this.uuid);
+
+  if(old == this.textContent)
+    this.textContent = old
+  else
+    instances.set(this.uuid, this.textContent)
```

Comment:
Mhh, good point, I was basing things on behavior I observed in my profile, where everything is fairly long-lived anyway.

The scenario you outline seems plausible for other types of script. I'll see If i can implement option a)

But for that one bit of information would be useful: Can I rely on all changes to `Script` instances notifying `service.config.addObserver()`